lib/galaxy/util/__init__.py

```diff
@@ -1360,16 +1362,16 @@ def size_to_bytes(size):
     """
     # Assume input in bytes if we can convert directly to an int
     try:
-        return int(size)
+        return int(float(size))
     except ValueError:
         pass
     # Otherwise it must have non-numeric characters
-    size_re = re.compile(r'([\d\.]+)\s*([eptgmk]b?|b|bytes?)$')
+    size_re = re.compile(r'(?P<size>[\d\.]+)\s*(?P<multiple>[eptgmk]?(b|bytes?)?)?$')
     size_match = size_re.match(size.lower())
     assert size_match is not None
-    size = float(size_match.group(1))
-    multiple = size_match.group(2)
-    if multiple.startswith('b'):
+    size = float(size_match.group("size"))
```

Comment:
Maybe call this `float_size` (also below) instead of reusing `size`, so that at the end we can add:
```python
    else:
        return size
```

Reply:
agreed (made it `number`). But the else branch should raise an exception which I added (the case of a size without a multiplier is treated in the 1st if branch).